examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -862,34 +1042,101 @@ def load_model_hook(models, input_dir):
+                                            "lr": args.text_encoder_lr if args.text_encoder_lr else args.learning_rate}
+        params_to_optimize = [unet_lora_parameters_with_lr, text_lora_parameters_one_with_lr,
+                              text_lora_parameters_two_with_lr]
+    else:
+        params_to_optimize = [unet_lora_parameters_with_lr]
+
+    # Optimizer creation
+    if args.use_8bit_adam and not args.optimizer.lower() == "adamw":
+        logger.warn(f"use_8bit_adam is ignored when optimizer is not set to 'AdamW'. Optimizer was "
+                    f"set to {args.optimizer.lower()}")
+
+    if args.optimizer.lower() == "prodigy":
         try:
-            import bitsandbytes as bnb
+            import prodigyopt
         except ImportError:
-            raise ImportError(
-                "To use 8-bit Adam, please install the bitsandbytes library: `pip install bitsandbytes`."
+            raise ImportError("To use Prodigy, please install the prodigyopt library: `pip install prodigyopt`")
+
+        optimizer_class = prodigyopt.Prodigy
+
+        if args.learning_rate <= 0.1:
+            logger.warn(
+                f"Learning rate is too low. When using prodigy, it's generally better to set learning rate around 1.0"
+            )
+        if args.text_encoder_lr:
+            logger.warn(
+                f"Learning rates were provided both for the unet and the text encdoer- e.g. text_encoder_lr and learning_rate"
+                f"when using prodigy only learning_rate is used as the initial learning rate"
             )
 
-        optimizer_class = bnb.optim.AdamW8bit
+        optimizer = optimizer_class(
+            params_to_optimize,
+            lr=args.learning_rate,
+            betas=(args.adam_beta1, args.adam_beta2),
+            beta3=args.prodigy_beta3,
+            weight_decay=args.adam_weight_decay,
+            eps=args.adam_epsilon,
+            decouple=args.prodigy_decouple,
+            use_bias_correction=args.prodigy_use_bias_correction,
+            safeguard_warmup=args.prodigy_safeguard_warmup,
+        )
+
+
+    elif args.optimizer.lower() == "adamw":
+        if args.use_8bit_adam:
+            try:
+                import bitsandbytes as bnb
+            except ImportError:
+                raise ImportError(
+                    "To use 8-bit Adam, please install the bitsandbytes library: `pip install bitsandbytes`."
+                )
+
+            optimizer_class = bnb.optim.AdamW8bit
+        else:
+            optimizer_class = torch.optim.AdamW
+
+        optimizer = optimizer_class(
+            params_to_optimize,
+            betas=(args.adam_beta1, args.adam_beta2),
+            weight_decay=args.adam_weight_decay,
+            eps=args.adam_epsilon,
+        )
     else:
-        optimizer_class = torch.optim.AdamW
+        raise ValueError(
+            f"Unsupported choice of optimizer: {args.optimizer.lower()}. Supported optimizers include [adamW, prodigy]")
```

Comment:
```suggestion
        logger.warn("f"Unsupported choice of optimizer: {args.optimizer}. Supported optimizers include [adamW, prodigy]. Defaulting to "adamW") 
        optimizer_class = torch.optim.AdamW
```

throwing an error here is to breaking IMO (even though we don't have backwards comp for `diffusers`. Let's insteal